dispatch/tests/test_api_imagegallery.py

```diff
@@ -0,0 +1,370 @@
+                "image_id": id1
+            },
+            {
+                "caption": "test caption 2",
+                "image_id": id2
+            }
+        ]
+
+        return (self._create_gallery_only(id, attachment), id1, id2)
+
+    def test_imagegallery_all(self):
+        """
+        Test that listing all the galleries works
+        """
+        # insert 2 galleries
+        gallery1, img_id11, img_id12 = self._create_gallery(0)
+        gallery2, img_id21, img_id22 = self._create_gallery(1)
+
+        url = reverse('api-galleries-list')
+        response = self.client.get(url, None, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+
+        # Check that the the returned data is correct
+        def checkOneGallery(id, img_id1, img_id2):
+            self.assertEqual(response.data['results'][id]['title'], 'Gallery Title %d' % id)
+            self.assertEqual(response.data['results'][id]['images'][0]['caption'], 'test caption 1')
+            self.assertEqual(response.data['results'][id]['images'][0]['image']['id'], img_id1)
+            self.assertEqual(response.data['results'][id]['images'][1]['caption'], 'test caption 2')
+            self.assertEqual(response.data['results'][id]['images'][1]['image']['id'], img_id2)
+
+        checkOneGallery(0, img_id11, img_id12)
+        checkOneGallery(1, img_id21, img_id22)
+
+    def test_imagegallery_get(self):
+        """
+        Test that getting a specific gallery works
+        """
+        gallery, img_id1, img_id2 = self._create_gallery(1)
+
+        id = gallery.data['id']
+        url = reverse('api-galleries-detail', args=[id])
+
+        response = self.client.get(url, None, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        self.assertEqual(response.data['title'], 'Gallery Title 1')
+        self.assertEqual(response.data['images'][0]['caption'], 'test caption 1')
+        self.assertEqual(response.data['images'][0]['image']['id'], img_id1)
+        self.assertEqual(response.data['images'][1]['caption'], 'test caption 2')
+        self.assertEqual(response.data['images'][1]['image']['id'], img_id2)
+
+    def test_imagegallery_get_invalidd(self):
+        """
+        Test that getting an invalid id returnsappropriate error
+        """
+        id = -1
+        url = reverse('api-galleries-detail', args=[id])
+        response = self.client.get(url, None, format='json')
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_imagegallery_create_unauthorized(self):
+        """
+        Create gallery should fail with unauthenticated request
+        """
+        # Clear authentication credentials
+        self.client.credentials()
+
+        url = reverse('api-galleries-list')
+        response = self.client.post(url, None, format='json')
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
```

Comment:
To make this test a bit more robust, you should send a valid payload containing gallery data and then check to verify that nothing was added to the database, like so:

```py
count = ImageGallery.objects.count()
self.assertEqual(count, 0)
```